app/controllers/comments_controller.rb

```diff
@@ -1,13 +1,42 @@
 class CommentsController < ApplicationController
   before_action :authenticate_user!
+
 def create
   @post = Post.find(params[:post_id])
   @post.comments.create(comment_params.merge(:user => current_user))
   redirect_to post_path(@post)
 end
 
+def edit
+  @comment = Comment.find(params[:comment_id])
+end
+
+def update
+  @comment = Comment.find(params[:comment_id])
+  @comment.update_attributes(comment_params)
+  redirect_to post_path(@post)
+end
+
+def destroy
+  @post = comment.post
+  comment.destroy 
+  redirect_to post_path(@post)
+end
+
 private
 
+def require_comment
+  render_not_found unless @comment
+end
+
+def comment
```

Comment:
You don't need to, but IMO it's a bit unclear that when you call `comment` it's a method, not a variable.  `comment` seems like an incredibly common variable name.